builder/amazon/common/run_config.go

```diff
@@ -141,6 +143,18 @@ func (c *RunConfig) Prepare(ctx *interpolate.Context) []error {
 		errs = append(errs, fmt.Errorf("shutdown_behavior only accepts 'stop' or 'terminate' values."))
 	}
 
+	if c.EnableT2Unlimited {
+		if c.SpotPrice != "" {
```

Comment:
Reviewing this just made me realise that about 4 years worth of my Packer scripts I thought were using spot instances actually weren't... I need to add a follow up to make Packer give a warning if `spot_price_auto_product` is set without `spot_price` being set to `auto`. 🤦‍♂️

Reply:
@jen20 If 6b532755599328d9e007dda1d80d4c2400969e9b isn't quite what you were after let me know and I'll make changes or drop as required. Thanks!

Reply:
Hi @DanHam - I didn't mean for you to do that as part of this work (which is fine as of 48afca2 - the failing test is unrelated to your change) - but that looks good to me too. Thanks!

Reply:
No problems - all good fun! Thanks for reviewing.